--- dspam-3.9.0/src/dspam.conf.in~	2010-01-11 22:21:58.000000000 +0100
+++ dspam-3.9.0/src/dspam.conf.in	2010-04-27 00:48:15.000000000 +0200
@@ -43,7 +43,7 @@
 # Other popular configurations:
 #TrustedDeliveryAgent "/usr/cyrus/bin/deliver"	# Cyrus
 #TrustedDeliveryAgent "/bin/maildrop"		# Maildrop
-#TrustedDeliveryAgent "/usr/local/sbin/exim -oMr spam-scanned" # Exim
+#TrustedDeliveryAgent "/usr/bin/exim -oMr spam-scanned" # Exim
 #
 TrustedDeliveryAgent "@delivery_agent@"
 
@@ -130,6 +130,7 @@
 Trust root
 Trust dspam
 Trust apache
+Trust http
 Trust mail
 Trust mailnull 
 Trust smmsp
@@ -778,7 +779,7 @@
 #ServerHost		127.0.0.1
 #ServerPort		24
 #ServerQueueSize	32
-#ServerPID		/var/run/dspam.pid
+ServerPID		/var/run/dspam/dspam.pid
 
 #
 # ServerMode specifies the type of LMTP server to start. This can be one of:
@@ -814,14 +815,14 @@
 # you are running the client and server on the same machine, as it eliminates
 # much of the bandwidth overhead.
 #
-#ServerDomainSocketPath	"/tmp/dspam.sock"
+ServerDomainSocketPath	"/var/run/dspam/dspam.sock"
 
 #
 # Client Mode: If you are running DSPAM in client/server mode, uncomment and
 # set these variables. A ClientHost beginning with a / will be treated as
 # a domain socket.
 #
-#ClientHost	/tmp/dspam.sock
+#ClientHost	/var/run/dspam/dspam.sock
 #ClientIdent	"secret@Relay1"
 #
 #ClientHost	127.0.0.1
