--- dspam-3.2.1/Makefile.in	2004-11-01 02:44:08.000000000 +0000
+++ dspam-3.2.1.new/Makefile.in	2004-11-01 12:42:07.487858984 +0000
@@ -322,7 +322,7 @@
 	read_config.c read_config.h 
 
 dspam_LDADD = libdspam.la
-dspam_LDFLAGS = -static
+dspam_LDFLAGS =
 ACLOCAL_AMFLAGS = -I m4
 sed_substitute_variables = \
         dspam_transformed=`echo dspam | sed '$(transform)'`; \
--- dspam-3.2.1/tools/Makefile.in	2004-11-01 02:44:07.000000000 +0000
+++ dspam-3.2.1.new/tools/Makefile.in	2004-11-01 12:43:19.931845824 +0000
@@ -229,17 +229,17 @@
 CLEANFILES = dspam_corpus dspam_genaliases
 bin_SCRIPTS = dspam_corpus dspam_genaliases
 dspam_2sql_SOURCES = dspam_2sql.c ../read_config.c
-dspam_2sql_LDFLAGS = -static
+dspam_2sql_LDFLAGS =
 dspam_dump_SOURCES = dspam_dump.c ../read_config.c
-dspam_dump_LDFLAGS = -static
+dspam_dump_LDFLAGS =
 dspam_clean_SOURCES = dspam_clean.c ../read_config.c
-dspam_clean_LDFLAGS = -static
+dspam_clean_LDFLAGS =
 dspam_crc_SOURCES = dspam_crc.c
-dspam_crc_LDFLAGS = -static
+dspam_crc_LDFLAGS =
 dspam_stats_SOURCES = dspam_stats.c ../read_config.c
 dspam_stats_LDFLAGS = -static
 dspam_merge_SOURCES = dspam_merge.c ../read_config.c
-dspam_merge_LDFLAGS = -static
+dspam_merge_LDFLAGS =
 dspam_admin_SOURCES = dspam_admin.c ../read_config.c
 dspam_admin_LDFLAGS = -static
 LDADD = ../libdspam.la
