--- dspam-3.0.0/Makefile.in	2004-06-14 14:25:43.000000000 +0000
+++ dspam-3.0.0.new/Makefile.in	2004-08-09 13:27:49.695629056 +0000
@@ -293,7 +293,7 @@
 	md5.c md5.h
 
 dspam_LDADD = libdspam.la
-dspam_LDFLAGS = -static
+dspam_LDFLAGS =
 ACLOCAL_AMFLAGS = -I m4
 all: auto-config.h
 	$(MAKE) $(AM_MAKEFLAGS) all-recursive
--- dspam-3.0.0/tools/Makefile.in	2004-06-14 14:25:45.000000000 +0000
+++ dspam-3.0.0.new/tools/Makefile.in	2004-08-09 13:29:11.738156696 +0000
@@ -222,16 +222,16 @@
 CLEANFILES = dspam_corpus dspam_genaliases
 bin_SCRIPTS = dspam_corpus dspam_genaliases
 dspam_2sql_SOURCES = dspam_2sql.c
-dspam_2sql_LDFLAGS = -static
+dspam_2sql_LDFLAGS =
 dspam_dump_SOURCES = dspam_dump.c
-dspam_dump_LDFLAGS = -static
+dspam_dump_LDFLAGS =
 dspam_clean_SOURCES = dspam_clean.c
-dspam_clean_LDFLAGS = -static
+dspam_clean_LDFLAGS =
 dspam_crc_SOURCES = dspam_crc.c
-dspam_crc_LDFLAGS = -static
+dspam_crc_LDFLAGS =
 dspam_stats_SOURCES = dspam_stats.c
 dspam_merge_SOURCES = dspam_merge.c
-dspam_merge_LDFLAGS = -static
+dspam_merge_LDFLAGS =
 LDADD = ../libdspam.la
 sed_substitute_variables = \
 	dspam_transformed=`echo dspam | sed '$(transform)'`; \
